src/Controls/src/Core/HandlerImpl/Application/Application.Impl.cs

```diff
@@ -111,10 +111,7 @@ public virtual void CloseWindow(Window window)
 
 		void IApplication.ThemeChanged()
 		{
-			if (UserAppTheme != AppTheme.Unspecified)
-				return;
-
-			TriggerThemeChangedActual();
+			PlatformAppTheme = AppInfo.RequestedTheme;
 		}
```

Comment:
Instead of being smart here, we just cache the new value from the OS and let the application decide what it wants to do. It is the same class, but now the property responds in a normal way to raise an event if the value has changed.